pkg/pwgen/pwrules/gen.go

```diff
@@ -99,12 +103,55 @@ func fetchRules() (map[string]jsonRule, error) {
 		return nil, err
 	}
 	var jr map[string]jsonRule
-	if err := json.NewDecoder(resp.Body).Decode(&jr); err != nil {
+	if err := json.NewDecoder(&cleaningReader{src: resp.Body}).Decode(&jr); err != nil {
 		return nil, err
 	}
 	return jr, nil
 }
 
+// TODO: remove when https://github.com/apple/password-manager-resources/pull/545 is merged
+type cleaningReader struct {
+	src io.Reader
+	rdr io.Reader
+}
+
+func (c *cleaningReader) init() error {
+	if c.rdr != nil {
+		return nil
+	}
+
+	var buf bytes.Buffer
+	scanner := bufio.NewScanner(c.src)
+	for scanner.Scan() {
+		line := scanner.Text()
+		skip := false
+		// skip two broken entries. this is a terrible hack because
+		// the JSON is not valid.
+		for _, needle := range []string{"fidelity.com", "hkexpress.com"} {
```

Comment:
Should we had a negative test that tests that `json.NewDecoder` fails when decoding these entries, so that once it won't fail any more we'll be able to remove the hardcoded exceptions??

Reply:
I don't have the motivation to do that, I'm afraid.

But feel free to add something.
But remember that we don't want to download the source on every build so it would need to be part the genrule

Reply:
I guess we could have a "monthly" GHA that checks if the data is still rotten or not.
I'll had an issue about it.